src/Security/Authorization/test/AuthorizationEndpointConventionBuilderExtensionsTests.cs

```diff
@@ -20,12 +20,11 @@ public void RequireAuthorization_IAuthorizeData()
         builder.RequireAuthorization(metadata);
 
         // Assert
-        var convention = Assert.Single(builder.Conventions);
-
         var endpointModel = new RouteEndpointBuilder((context) => Task.CompletedTask, RoutePatternFactory.Parse("/"), 0);
+        var convention = Assert.Single(builder.Conventions);
         convention(endpointModel);
 
-        Assert.Equal(metadata, Assert.Single(endpointModel.Metadata));
+        Assert.Equal(metadata, Assert.Single(endpointModel.Metadata.Where(t => t is IAuthorizeData)));
```

Comment:
I'm guessing this was changed because of the previous cache in metadata. Is it necessary now?

Reply:
Revert throughout PR if not necessary

Reply:
Maybe all the changes in AuthorizationEndpointConventionBuilderExtensionsTests.cs should be undone?

Reply:
Yeah can revert this file good call